compiler/src/dotty/tools/dotc/core/tasty/AttributeUnpickler.scala

```diff
@@ -0,0 +1,28 @@
+package dotty.tools.dotc
+package core.tasty
+
+import scala.language.unsafeNulls
+
+import dotty.tools.tasty.{TastyFormat, TastyReader, TastyBuffer}
+
+import java.nio.charset.StandardCharsets
+
+class AttributeUnpickler(reader: TastyReader):
+  import reader._
+
+  private[tasty] lazy val attributes: List[String] = {
+    val attributesBuilder = List.newBuilder[String]
+    while (!isAtEnd) {
+      val length = readNat()
+      if (length > 0) {
```

Comment:
This test probably should not be there. It would most likely be invalid to have a zero length, but if it is not, it means there is actually an empty string attribute.

Reply:
I will specify and validate that attributes are non-empty strings.